cdc_replication_test.py

```diff
@@ -0,0 +1,193 @@
+        print("Waiting for added node to initialize...")
+        self.db_cluster.wait_for_init(node_list=[node3])
+
+        master_node = self.db_cluster.nodes[0]
+        replica_node = self.cs_db_cluster.nodes[0]
+
+        print('Creating table on master cluster.')
+        with self.cql_connection_patient(node=master_node) as sess:
+            sess.execute(keyspace_def(3))
+            sess.execute(TABLE_DEF_MASTER)
+
+        print('Creating table on replica cluster.')
+        with self.cql_connection_patient(node=replica_node) as sess:
+            sess.execute(keyspace_def(1))
+            sess.execute(TABLE_DEF_BASE)
+
+        print('Waiting 40s for the latest CDC generation to start...')
+        time.sleep(40)
+
+        print("Starting cassandra-stress")
+        stress_thread = self.run_stress_cassandra_thread(stress_cmd=self.params.get('stress_cmd'))
+
+        print('Let C-S run for a while...')
+        time.sleep(10)
+
+        print("Starting nemesis.")
+        self.db_cluster.add_nemesis(nemesis=self.get_nemesis_class(), tester_obj=self)
+        self.db_cluster.start_nemesis()
+
+        loader_node : BaseNode = self.loaders.nodes[0]
+
+        print('Installing tmux on loader node.')
+        res = loader_node.remoter.run(cmd='sudo yum install -y tmux')
+        if res.exit_status != 0:
+            self.fail('Could not install tmux.')
+
+        print('Getting scylla-migrate on loader node.')
+        res = loader_node.remoter.run(cmd=f'wget {SCYLLA_MIGRATE_URL} -O scylla-migrate && chmod +x scylla-migrate')
+        if res.exit_status != 0:
+            self.fail('Could not obtain scylla-migrate.')
+
+        print('Getting replicator on loader node.')
+        res = loader_node.remoter.run(cmd=f'wget {REPLICATOR_URL} -O replicator.jar')
+        if res.exit_status != 0:
+            self.fail('Could not obtain CDC replicator.')
+
+        # We run the replicator in a tmux session so that remoter.run returns immediately
+        # (the replicator will run in the background). We redirect the output to a log file for later extraction.
+        replicator_script = dedent("""
+            (cat >runreplicator.sh && chmod +x runreplicator.sh && tmux new-session -d -s 'replicator' ./runreplicator.sh) <<'EOF'
+            #!/bin/bash
+            
+            java -cp replicator.jar com.scylladb.scylla.cdc.replicator.Main -k ks -t tb -s {} -d {} -cl {} 2>&1 | tee replicatorlog
+            EOF
+        """.format(master_node.external_address, replica_node.external_address, 'one'))
+
+        print('Replicator script:\n', replicator_script)
+
+        print('Starting replicator.')
+        res = loader_node.remoter.run(cmd=replicator_script)
+        if res.exit_status != 0:
+            self.fail('Could not start CDC replicator.')
+
+        print("Let C-S run for a while...")
+        time.sleep(30)
+
+        print("Stopping nemesis before bootstrapping a new node.")
+        self.db_cluster.stop_nemesis(timeout=60)
```

Comment:
if you plan to use other nemesis than RandomInterruptionNetworkMonkey, you may need to wait for the nemesis to actually finish.

Reply:
I saw in the code that `stop_nemesis` joins all nemesis threads. What else do I need to do?

Reply:
this is o.k. usually we want more time, since some Nemesis might take longer then 60sec
I think we wait 30min.